src/rust/engine/fs/src/lib.rs

```diff
@@ -353,17 +353,67 @@ fn is_ignored<P: AsRef<Path>>(ignore: &Gitignore, path: P, is_dir: bool) -> bool
 }
 
 ///
+/// A wrapper around a CpuPool, to add the ability to drop the pool before forking,
+/// and then lazily re-initialize it in a new process.
+///
+pub struct ResettablePool {
+  name_prefix: String,
+  pool: RwLock<Option<CpuPool>>,
+}
+
+impl ResettablePool {
+  pub fn new(name_prefix: String) -> ResettablePool {
+    ResettablePool {
+      name_prefix: name_prefix,
+      pool: RwLock::new(None),
+    }
+  }
+
+  pub fn with<T, F: FnOnce(&CpuPool) -> T>(&self, f: F) -> T {
```

Comment:
I'm vaguely inclined to just implement spawn_fn here, which delegates to the underlying CpuPool, rather than exposing a wrapper around the underlying CpuPool. But I can easily be swayed the other way too.